Documentation/sources/content-card-ui/public-classes/aepui.md

```diff
@@ -0,0 +1,96 @@
+# Sealed Interface - AepUI
+
+The `AepUI` interface represents a UI component that can be rendered using the AEP compose UI library. In the initial release of the AEP compose UI library, the `SmallImageUI` implements an `AepUI`  to display a Small Image Template. The `AepUI` associates with a specific template and state to ensure consistency when rendering and managing UI components.
+## Methods
+
+### getTemplate 
+
+Retrieves the template associated with this UI component.
+
+#### Returns
+
+A template of type `T` associated with the UI componenet.
+
+#### Syntax
+
+``` kotlin
+fun getTemplate(): T
+```
+
+### getState 
+
+Retrieves the current state of the UI component.
+
+#### Returns
+
+A state of type `S` associated with the UI componenet.
+
+#### Syntax
+
+``` kotlin
+fun getState(): S
+```
+
+### updateState 
+
+Updates the state of the UI component with a new state.
+
+#### Parameters
+
+- _newState_ - The new state of type `S` to update within the UI component.
+
+#### Syntax
+
+``` kotlin
+fun updateState(newState: S)
+```
+
+# Implementing Classes
+
+## Class - SmallImageUI
+
+Class representing a small image AEP UI, which implements the [AepUI](./aepui.md#Sealed Interface - AepUI) interface.
```

Comment:
nit: Do you think its helpful to tie to back to which template it is used to create the UI for? Something along this  line : `Implementation of AepUI to support rendering UI for SmallImageTemplate`
Please feel free to also update the kdocs if this makes more sense

Reply:
this makes sense. will update in both areas.